spec/models/reports/facility_state_spec.rb

```diff
@@ -259,6 +259,232 @@
+          _non_diabetic_patients = create_list(:patient, 3, :without_diabetes, registration_facility: facility, recorded_at: june_2021[:two_months_ago])
+
+          month_2020_07 = "2020-07-01"
+          month_2021_04 = "2021-04-01"
+
+          refresh_views
+          with_reporting_time_zone do
+            expect(described_class
+                     .where(facility_id: facility.id)
+                     .where("month_date < ?", month_2020_07)
+                     .pluck(:monthly_diabetes_registrations)).to all be_nil
+
+            expect(
+              described_class
+                .where(facility_id: facility.id)
+                .where(month_date: month_2020_07)
+                .pluck(:monthly_diabetes_registrations)
+            ).to eq [old_diabetes_patients.size]
+
+            expect(
+              described_class
+                .where(facility_id: facility.id)
+                .where("month_date > ?", month_2020_07)
+                .where("month_date < ?", month_2021_04)
+                .pluck(:monthly_diabetes_registrations)
+            ).to all eq 0
+
+            expect(
+              described_class
+                .where(facility_id: facility.id)
+                .where(month_date: month_2021_04)
+                .pluck(:monthly_diabetes_registrations)
+            ).to eq [new_diabetic_patients.size]
+          end
+        end
+      end
+    end
+
+    context "assigned patients by care states" do
+      describe "under care, lost to follow up and dead patients" do
+        it "has the number of assigned diabetic patients which are under care,  lost to follow up or dead" do
+          facility = create(:facility)
+
+          # non diabetic patient
+          _non_diabetic_patient = create(:patient, :without_diabetes, assigned_facility: facility)
+
+          # ltfu diabetes patient
+          ltfu_diabetic_patients = create_list(:patient, 2, :diabetes, assigned_facility: facility, recorded_at: june_2021[:over_12_months_ago])
+          create(:blood_sugar, :with_encounter, patient: ltfu_diabetic_patients.first, recorded_at: june_2021[:over_12_months_ago])
+
+          # under care diabetes patients
+          under_care_diabetic_patients = create_list(:patient, 3, :diabetes, assigned_facility: facility, recorded_at: june_2021[:over_12_months_ago])
+          create(:blood_sugar, :with_encounter, patient: under_care_diabetic_patients.first, recorded_at: june_2021[:one_month_ago])
+          create(:blood_pressure, :with_encounter, patient: under_care_diabetic_patients.second, recorded_at: june_2021[:one_month_ago])
+          create(:appointment, patient: under_care_diabetic_patients.third, recorded_at: june_2021[:one_month_ago])
+
+          # dead patients
+          dead_diabetic_patients = create_list(:patient, 3, :diabetes, status: "dead", assigned_facility: facility, recorded_at: june_2021[:under_3_months_ago])
```

Comment:
Let's make sure the number of dead patients is different from the other two categories. Otherwise this test is susceptible to a false positive if we accidentally swap indicators.

Can we take care to do that throughout this spec file?

Reply:
I've updated all the diabetes specs to also check that there are no false positives.

Reply:
Those are good specs, but my concern was about the diabetic patients in here. Can we make the number of dead patients `4`? If both dead and under care are `3` patients in this spec, then our test won't fail if we accidentally swap those two. It's unlikely but very easy to mitigate.